clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -366,8 +383,16 @@ public void close() {
         }
     }
 
+    // visible for testing
+    void invokeCompletedOffsetCommitCallbacks() {
+        for (OffsetCommitCompletion completion : completedOffsetCommits)
+            completion.invoke();
```

Comment:
We could probably change to an iterator and remove each completion before invoking.